### PR TITLE
tests for some transitions between states of the exit

### LINKED ISSUE #23: add more tests
From @pdobacz:

> a batch of longer-running tests that test whether gas is always under control for submitBlock, startExit, challengeExit and finalizeExit, no matter how much we put into the contract, like we did for HonteStaking.sol

> Another, slightly paranoid thing I'd consider is adding assertions on pre/post-conditions to actions executed using testlang, like "if a successful deposit is done, then contract's balance increases, depositor's - decreases", "if a successful finalizeExit is done, then balances are updated properly". Likewise, I'd assert proper event firing in post conditions. This is something we did for HonteStaking too.

From @pgebal:
> fix `leaves=[]` in FixedMerkle
> ~add a test for a failure of exiting utxo that is already exited (finalized)~ will do in #19

### PR DESCRIPTION
**Pull request by paulperegud**
closes #23

### CODE REVIEW

File: tests/contracts/root_chain/test_process_exits.py [hunk 1]
```diff
@@ -150,15 +150,28 @@ def test_finalize_exits_tx_race(testlang):
...
+    testlang.start_standard_exit(utxo4.owner, utxo4.spend_id)
+
+    testlang.forward_timestamp(2 * WEEK + 1)
+    testlang.finalize_exits(NULL_ADDRESS, utxo1.spend_id, 1)
+
     testlang.finalize_exits(NULL_ADDRESS, utxo2.spend_id, 3)
-    gas2 = testlang.ethtester.chain.last_gas_used()
-    assert gas1 < 4000
-    assert gas2 > 30000
+    three_exits_gas = testlang.ethtester.chain.last_gas_used()
```
Comment (DavidKnott) [inline tests/contracts/root_chain/test_process_exits.py]: @paulperegud Doesn't `last_gas_used` only get the gas used from the last transaction as opposed to the last three exits?

File: testlang/testlang.py [hunk 2]
```diff
@@ -338,17 +345,24 @@ def get_standard_exit(self, utxo_id):
...
             int: The account's balance.
         """
-
-        return self.ethtester.chain.head_state.get_balance(account.address)
+        if token == NULL_ADDRESS:
+            return self.ethtester.chain.head_state.get_balance(account.address)
+        if hasattr(token, "balanceOf"):
+            return token.balanceOf(account.address)
+        else:
+            token_contract = conftest.watch_contract(self.ethtester, 'MintableToken', token)
```
Comment (DavidKnott) [inline testlang/testlang.py]: What's special about `MintableToken` does it not have `balanceOf`?

Comment (paulperegud) [reply testlang/testlang.py]: This is misleading. Token can be an address or a wrapped contract.

File: tests/contracts/root_chain/test_process_exits.py (hunk 1 above)
Comment (paulperegud) [reply tests/contracts/root_chain/test_process_exits.py]: Those three exits are being done as single Ethereum transaction. See third parameter in finalize_exits call.

File: testlang/testlang.py (hunk 2 above)
Comment (paulperegud) [reply testlang/testlang.py]: Would love to be able to change that.

**Pull request merged**